### PR TITLE
fix(spellcheck): fix several spellcheck issues not caught in CI

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes some issus later found by CI that somehow didn't block the originating PRs.

Example failure:

https://github.com/emqx/emqx/actions/runs/5715470851/job/15485284918?pr=11372

### <samp>🤖 Generated by Copilot at b832b6e</samp>

This pull request improves the documentation and UI of the `emqx_bridge_azure_event_hub` application by adding and fixing descriptions and labels for various configuration fields and structs. It also updates the version number of the application, fixes some typos in other bridge applications, and enhances the spellcheck script by enabling interactive mode.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_bridge_azure_event_hub/src/emqx_bridge_azure_event_hub.erl [hunk 1]
```diff
@@ -162,7 +173,15 @@ ref(Name) ->
...
 producer_overrides() ->
     #{
-        authentication => mk(ref(auth_username_password), #{default => #{}, required => true}),
+        authentication =>
+            mk(
+                ref(auth_username_password),
+                #{
+                    default => #{},
+                    required => true,
+                    desc => ?DESC(auth_sasl_password)
```
Comment (paulozulato) [inline apps/emqx_bridge_azure_event_hub/src/emqx_bridge_azure_event_hub.erl:182]: the DESC here should be `auth_sasl_password` or `auth_username_password`?

File: rel/i18n/emqx_bridge_azure_event_hub.hocon [hunk 2]
```diff
@@ -284,4 +284,28 @@ sync_query_timeout.desc:
...
 """Synchronous Query Timeout"""
 
+auth_username_password.desc:
+"""Username/password based authentication."""
+
+auth_username_password.label:
+"""Username/password Auth"""
+
+auth_sasl_password.desc:
+"""The password for connecting to Azure Event Hub.  Should be the \"connection string-primary key\" of a Namespace shared access policy."""
```
Comment (paulozulato) [inline rel/i18n/emqx_bridge_azure_event_hub.hocon:294]:
```suggestion
"""The password for connecting to Azure Event Hub. Should be the "connection string-primary key" of a Namespace shared access policy."""
```

**Review by id — DISMISSED** [review]

File: apps/emqx_bridge_azure_event_hub/src/emqx_bridge_azure_event_hub.erl [hunk 3]
```diff
@@ -162,7 +173,15 @@ ref(Name) ->
 
 producer_overrides() ->
     #{
-        authentication => mk(ref(auth_username_password), #{default => #{}, required => true}),
+        authentication =>
+            mk(
+                ref(auth_username_password),
+                #{
+                    default => #{},
```
[the remark this answers is not in the document]
Comment (thalesmg) [reply apps/emqx_bridge_azure_event_hub/src/emqx_bridge_azure_event_hub.erl:180]: I want to override an [existing default](https://github.com/emqx/emqx/blob/2a29d2a141cd2e0e5d843a13e128417ce4d0ab97/apps/emqx_bridge_kafka/src/emqx_bridge_kafka.erl#L200) here.

File: apps/emqx_bridge_azure_event_hub/src/emqx_bridge_azure_event_hub.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_bridge_azure_event_hub/src/emqx_bridge_azure_event_hub.erl:182]: It should actually be `"authentication"` here.

**Review by qzhuyan — APPROVED** [review]

**Review by paulozulato — APPROVED** [review]

**Pull request merged**